src/devices/Pn532/Pn532.cs

```diff
@@ -0,0 +1,1623 @@
+        /// <param name="dataFromCard">The potential data to receive</param>
+        /// <returns>The number of bytes read</returns>
+        public override int WriteRead(byte targetNumber, ReadOnlySpan<byte> dataToSend, Span<byte> dataFromCard)
+        {
+            Span<byte> toSend = stackalloc byte[1 + dataToSend.Length];
+            toSend[0] = targetNumber;
+            if (dataToSend.Length > 0)
+                dataToSend.CopyTo(toSend.Slice(1));
+            var ret = WriteCommand(CommandSet.InDataExchange, toSend);
+            if (ret < 0) return -1;
+            Span<byte> toReceive = stackalloc byte[1 + dataFromCard.Length];
+            ret = ReadResponse(CommandSet.InDataExchange, toReceive);
+            toReceive.Slice(1).CopyTo(dataFromCard);
+            if ((toReceive[0] == (byte)ErrorCode.None) & (ret >= 0))
+                return ret;
+            return -1;
+        }
+
+        /// <summary>
+        /// Automatically poll specific types of devices
+        /// </summary>
+        /// <param name="numberPolling">The number of polling before accepting a card</param>
+        /// <param name="periodMilliSecond">The period of polling before accepting a card</param>
+        /// <param name="pollingType">The type of cards to poll</param>
+        /// <returns>A raw byte array containing the number of cards, the card type and the raw data. Null if nothing hasd been polled</returns>
+        public byte[] AutoPoll(byte numberPolling, ushort periodMilliSecond, PollingType[] pollingType)
+        {
+            if (pollingType == null) return null;
+            if (pollingType.Length > 15) return null;
+            Span<byte> toSend = stackalloc byte[2 + pollingType.Length];
+            toSend[0] = numberPolling;
+            if ((periodMilliSecond / 150) > 0xFF)
+                toSend[1] = 0xFF;
+            else
+                toSend[1] = (byte)(periodMilliSecond / 150);
+            for (int i = 0; i < pollingType.Length; i++)
+                toSend[2 + i] = (byte)pollingType[i];
+            var ret = WriteCommand(CommandSet.InAutoPoll, toSend);
+            if (ret < 0) return null;
+            Span<byte> receivedData = stackalloc byte[1024];
+            ret = ReadResponse(CommandSet.InAutoPoll, receivedData);
+            LogInfo.Log($"{nameof(AutoPoll)}, success: {ret}", LogLevel.Debug);
+            if (ret >= 0)
+            {
+                return receivedData.Slice(0, ret).ToArray();
+            }
+            return null;
+        }
+
+
+#region RFConfiguration
+
+        /// <summary>
+        /// Set the Radio Frequence Field Mode
+        /// </summary>
+        /// <param name="rfFieldMode">Radio Frequence Field Mode</param>
+        /// <returns>True is success</returns>
+        public bool SetRfField(RfFieldMode rfFieldMode)
+        {
+            return RfConfiguration(RfConfigurationMode.RfField, new byte[1] { (byte)rfFieldMode });
```

Comment:
RfConfiguration should have a verb in the name, otherwise it's weird to read this code